File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for(int num: nums) {
            seen.add(num);
        }
        int max = 0;
        for(int num:seen) {
            if(!seen.contains(num-1)) {
                int streak = 1;
                int curr = num;
                while(seen.contains(curr+1)) {
                    streak++;
                    curr++;
                }
                max = Math.max(max, streak);
            }
        }

        return max;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max = 0
        start,end = 0,1
        if len(prices) <= 1:
            return 0
        while end < len(prices):
            if prices[end] - prices[start] > max:
                max = prices[end] - prices[start]
            if prices[start] > prices[end]:
                start = end
            end += 1
        return max
        